src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapResolvedState.java

```diff
@@ -0,0 +1,200 @@
+
+    Assert.assertTrue(vuap.isResolvedStateDisplayed(defaultVideoHeight, vuap.getVideoHieght()));
+  }
+
+  @Test(groups = "AdsVuapResolvedStateAutoplayOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapResolvedStateDesktop")
+  public void vuapResolvedStateShouldStartPlayingAdvertisementAutomatically(Page page, String slot, String videoIframeSelector) {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+    ads.refreshPage();
+
+    verifyVideAutoplay(vuap);
+  }
+
+  @Test(groups = "AdsVuapResolvedStateTimeProgressOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapResolvedStateDesktop")
+  public void vuapResolvedStateShouldProgressInTime(Page page, String slot, String videoIframeSelector) {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    verifyVideoTimeIsProgressing(vuap);
+  }
+
+  @Test(groups = "AdsVuapResolvedStateClickOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapResolvedStateDesktop")
+  public void vuapResolvedStateShouldHaveLinkToFandomOnImage(Page page, String slot, String videoIframeSelector) {
+    final String expected = "http://www.wikia.com/fandom";
+
+    final AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    vuap.clickOnImageResolvedState();
+
+    final String actual = ads.switchToNewBrowserTab();
+    Assert.assertTrue(actual.equals(expected), "Image should point to page on fandom.");
+  }
+
+  @Test(groups = "AdsVuapResolvedStateMuteOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapResolvedStateDesktop")
+  public void vuapResolvedStateShouldMute(Page page, String slot, String videoIframeSelector) {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    verifyAutoplayIsMuted(vuap);
+  }
+
+  @Test(groups = "AdsVuapResolvedEndOasis",
+      dataProviderClass = AdsDataProvider.class,
+      dataProvider = "adsVuapResolvedStateDesktop")
+  public void vuapResolvedShouldEnd(Page page, String slot, String videoIframeSelector) {
+    AdsBaseObject ads = new AdsBaseObject(driver, urlBuilder.getUrlForPage(page), DESKTOP_SIZE);
+    final AutoplayVuap vuap = new AutoplayVuap(driver, slot, videoIframeSelector);
+
+    ads.refreshPage();
+
+    verifyVideEndedAndReplyButtonDisplayed(vuap);
+  }
+
+  private void verifyAutoplayIsMuted(final AutoplayVuap vuap) {
```

Comment:
shouldn't it be a part of AutoplayVuap class or a different class with assertions related to this?